hail/python/hailtop/batch/backend.py

```diff
@@ -543,8 +543,14 @@ def _run(self,
         token:
             If not `None`, a string used for idempotency of batch submission.
         """
-        return async_to_blocking(
-            self._async_run(batch, dry_run, verbose, delete_scratch_on_exit, wait, open, disable_progress_bar, callback, token, **backend_kwargs))
+        try:
+            return async_to_blocking(
+                self._async_run(batch, dry_run, verbose, delete_scratch_on_exit, wait, open, disable_progress_bar, callback, token, **backend_kwargs))
+        except KeyboardInterrupt:
+            if batch._batch_handle is not None:
+                print("Received a keyboard interrupt, cancelling the batch...")
+                batch._batch_handle.cancel()
+            raise
```

Comment:
I tried to make this change a long time ago and got a bunch of push back.

cc: @konradjk ; If you ctrl-c a `hailtop.batch` script that is waiting on a batch (e.g. `b.run(wait=True)`), should the batch be cancelled?

Reply:
For now, let's just make the QoB change and revisit this after some discussion.

Reply:
for context: https://hail.zulipchat.com/#narrow/stream/127527-team/topic/0.2E2.2E78/near/257583824

Reply:
Interesting, I remember the argument but feel like these two behaviors are consistent with each other. Happy to remove the batch behavior for now though.

Reply:
I completely agree with you but, as the thread suggests, there was divergent opinions. I think people think of Batch as a submission sort of thing. IMHO, if you're trying to submit you should `wait=False`, but 🤷 . Let's take the win and fight this battle separately.